Purchases/SubscriberAttributes/RCPurchases+SubscriberAttributes.m

```diff
@@ -50,8 +51,16 @@ - (RCSubscriberAttributeDict)unsyncedAttributesByKey {
     return [self.subscriberAttributesManager unsyncedAttributesByKeyForAppUserID:self.appUserID];
 }
 
-- (void)markAttributesAsSynced:(RCSubscriberAttributeDict)syncedAttributes
-                     appUserID:(NSString *)appUserID {
+- (void)markAttributesAsSyncedIfNeeded:(RCSubscriberAttributeDict)syncedAttributes
```

Comment:
unified the logic of "should I mark synced?" + "log attributes errors" here.

Reply:
if we're already printing out the user info on post receipt, then this might issue a duplicate log. I feel like that might not be a big deal, and I couldn't find where we might be logging the user info.